aspnetcore/fundamentals/portable-object-localization.md

```diff
@@ -0,0 +1,218 @@
+a cardinality of one is mapped to the first plural form. Any other cardinality is mapped to the second plural form.
+
+Unfortunately, not all languages share the same rules. This is illustrated with the Czech language, which has three plural forms.
+
+Create the `cs.po` file as follows, and note how the pluralization needs three different translations:
+
+[!code-text[Main](localization/sample/POLocalization/cs.po)]
+
+To accept Czech localizations, add `"cs"` to the list of supported cultures in the `ConfigureServices` method of *Startup.cs*:
+
+```csharp
+var supportedCultures = new List<CultureInfo>
+{
+    new CultureInfo("en-US"),
+    new CultureInfo("en"),
+    new CultureInfo("fr-FR"),
+    new CultureInfo("fr"),
+    new CultureInfo("cs")
+};
+```
+
+Edit the *Views/Home/About.cshtml* file to render localized, plural strings for several cardinalities:
+
+```cshtml
+<p>@Localizer.Plural(1, "There is one item.", "There are {0} items.")</p>
+<p>@Localizer.Plural(2, "There is one item.", "There are {0} items.")</p>
+<p>@Localizer.Plural(5, "There is one item.", "There are {0} items.")</p>
+```
+
+**Note:** In a real world scenario, a variable would be used to represent the count. Here, we repeat the same code with three different values to expose a very specific case.
+
+Upon switching cultures, you see the following:
+
+For `/Home/About`:
+
+```html
+There is one item.
+There are 2 items.
+There are 5 items.
+```
+
+For `/Home/About?culture=fr`:
+
+```html
+Il y a un élément.
+Il y a 2 éléments.
+Il y a 5 éléments.
+```
+
+For `/Home/About?culture=cs`:
+
+```html
+Existuje jedna položka.
+Existují 2 položky.
+Existuje 5 položek.
+```
+
+Note that for the Czech culture, the three translations are different. The French and English cultures share the same construction for the two last translated strings.
+
+## Advanced tasks
+
+### Contextualizing strings
+
+Applications often contain the same strings to be translated in several places, while requiring the flexibility to define different translations. A PO file supports the notion of a *context*, which can be used to categorize the string being represented.
+
+The PO localization services can use the name of the full class or the view that is used when translating a string. This is accomplished by setting the value on the `msgctx` entry.
+
+Considering the previous example, the entry could have been written as:
+
+*fr.po*
+
+```text
+msgctx Views.Home.About
+msgid "Hello world!"
+msgstr "Bonjour le monde!"
+```
+
+When no context is specified, it's used as a fallback for any string whose context doesn't match any specific one.
+
+### Changing the location of PO files
+
+You can change the default location of PO files in the application by setting a base folder name. This is accomplished with the following code in the `ConfigureServices` method of *Startup.cs*:
```

Comment:
Get rid of you.
The default location of PO files can be changed in `ConfigureServices`: (Don't tell them it's in starup.cs, this is not a beginning tutorial).